essentials-base/src/main/kotlin/io/github/nyliummc/essentials/commands/FlightCommand.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) NyliumMC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.github.nyliummc.essentials.commands
+
+import com.mojang.brigadier.CommandDispatcher
+import com.mojang.brigadier.context.CommandContext
+import io.github.ladysnake.pal.Pal
+import io.github.ladysnake.pal.VanillaAbilities
+import io.github.nyliummc.essentials.api.builders.Command
+import io.github.nyliummc.essentials.api.builders.Text
+import net.minecraft.command.arguments.EntityArgumentType
+import net.minecraft.server.command.ServerCommandSource
+import net.minecraft.server.network.ServerPlayerEntity
+
+object FlightCommand {
+    fun register(dispatcher: CommandDispatcher<ServerCommandSource>) {
+        Command.builder(dispatcher) {
+            command("flight", "fly") {
+                executes (::toggleFlightSelf)
+                argument("player", EntityArgumentType.player()) {
+                    executes (::toggleFlightOther)
+                }
+            }
+        }
+    }
+
+    private fun toggleFlightSelf(commandContext: CommandContext<ServerCommandSource>): Int {
+        // Get player
+        val player : ServerPlayerEntity = commandContext.source.player
+
+        // Set flight
+        if (player.abilities.allowFlying) {
+            Pal.revokeAbility(player, VanillaAbilities.ALLOW_FLYING, Pal.getAbilitySource("essentials", "source"));
+            Pal.revokeAbility(player, VanillaAbilities.FLYING, Pal.getAbilitySource("essentials", "source"));
+        } else {
+            Pal.grantAbility(player, VanillaAbilities.ALLOW_FLYING, Pal.getAbilitySource("essentials", "source"));
+            Pal.grantAbility(player, VanillaAbilities.FLYING, Pal.getAbilitySource("essentials", "source"));
+        }
```

Comment:
Why not make this logic a common private method and keep the feedback and getting the player different between the two executors.